### ISSUE TITLE
Early implementation of FuzzyMatching 

### ISSUE THREAD

**Issue by peterjamesnugent** [BHoM/AGS_Toolkit#3]
#### Description:

Add [FuzzySharp](https://github.com/JakeBayer/FuzzySharp) functionality to enable usage within the toolkit. 

Before merging, we eeed to make a decision for:
- Where does FuzzySharp sit? 
1. Within FuzzyToolkit (won't be able to use within the BHoM_Engine or BHoM_UI)
2. Within BHoM_Engine (under Fuzzy_Engine or perhaps sitting within the Data_Engine)

Some thought needs to be given to the actual implementation - whether it is just a reflection of the different ratio methods, or whether there needs to be a BHoM specific implementation relevant to our uses.

**Comment by FraserGreenroyd** [comment]
Data Engine gets my vote 👍

**Comment by alelom** [comment]
I think I proposed this in the past too – I'd love to have a **Text_Engine** where we could place things like Fuzzy, Levenshtein distance, other [string metric](https://en.wikipedia.org/wiki/String_metric) functions, RegEx based engine functions, and similar. 

I would personally argue against having a dedicated FuzzyToolkit or Fuzzy_Engine because it would be excessively specific. 

On the other hand, I think that the integration of Fuzzy in an existing Engine may be ok, although I personally wouldn't think of looking for text metric methods in the Data_Engine (or one of the other already existing engines, for that matter).

**Comment by al-fisher** [comment]
Agreed - sounding good adding to core BHoM_Engine.

I also like Text_Engine - makes sense.

**Comment by peterjamesnugent** [comment]
I think we have general agreement that a separate toolkit is in appropriate given that we want access to these functions within the core BHoM_Engine.

We had a general discussion today and there's consensus that there are use cases for it across the different namespaces, particularly in the Revit space (where labelling may not always be consistent), searching datasets (in Structures) etc.

We did start to converge on Search_Engine for a name, because a lot of focus was on the actual search, rather than text processing. We've scheduled another catchup for 22nd January 2024 at 1600 GMT, in the meantime I will continue to prototype on the AGS_Toolkit.
